src/falcon/falcon_c/rpo.c

```diff
@@ -0,0 +1,757 @@
+        14200078843431360086ULL,
+    },
+    {
+        6202948458916099932ULL,
+        17690140365333231091ULL,
+        3595001575307484651ULL,
+        373995945117666487ULL,
+        1235734395091296013ULL,
+        14172757457833931602ULL,
+        707573103686350224ULL,
+        15453217512188187135ULL,
+        219777875004506018ULL,
+        17876696346199469008ULL,
+        17731621626449383378ULL,
+        2897136237748376248ULL,
+    },
+    {
+        8023374565629191455ULL,
+        15013690343205953430ULL,
+        4485500052507912973ULL,
+        12489737547229155153ULL,
+        9500452585969030576ULL,
+        2054001340201038870ULL,
+        12420704059284934186ULL,
+        355990932618543755ULL,
+        9071225051243523860ULL,
+        12766199826003448536ULL,
+        9045979173463556963ULL,
+        12934431667190679898ULL,
+    },
+    {
+        18389244934624494276ULL,
+        16731736864863925227ULL,
+        4440209734760478192ULL,
+        17208448209698888938ULL,
+        8739495587021565984ULL,
+        17000774922218161967ULL,
+        13533282547195532087ULL,
+        525402848358706231ULL,
+        16987541523062161972ULL,
+        5466806524462797102ULL,
+        14512769585918244983ULL,
+        10973956031244051118ULL,
+    },
+    {
+        6982293561042362913ULL,
+        14065426295947720331ULL,
+        16451845770444974180ULL,
+        7139138592091306727ULL,
+        9012006439959783127ULL,
+        14619614108529063361ULL,
+        1394813199588124371ULL,
+        4635111139507788575ULL,
+        16217473952264203365ULL,
+        10782018226466330683ULL,
+        6844229992533662050ULL,
+        7446486531695178711ULL,
+    },
+    {
+        3736792340494631448ULL,
+        577852220195055341ULL,
+        6689998335515779805ULL,
+        13886063479078013492ULL,
+        14358505101923202168ULL,
+        7744142531772274164ULL,
+        16135070735728404443ULL,
+        12290902521256031137ULL,
+        12059913662657709804ULL,
+        16456018495793751911ULL,
+        4571485474751953524ULL,
+        17200392109565783176ULL,
+    },
+    {
+        17130398059294018733ULL,
+        519782857322261988ULL,
+        9625384390925085478ULL,
+        1664893052631119222ULL,
+        7629576092524553570ULL,
+        3485239601103661425ULL,
+        9755891797164033838ULL,
+        15218148195153269027ULL,
+        16460604813734957368ULL,
+        9643968136937729763ULL,
+        3611348709641382851ULL,
+        18256379591337759196ULL,
+    },
+};
+
+/*
+ * Define functions
+ */
+
+void apply_round(uint64_t *const state,
+                 const uint64_t round);
+
+void apply_sbox(uint64_t *const state);
+
+void apply_mds(uint64_t *state);
+
+void apply_constants(uint64_t *const state,
+                     const uint64_t *ark);
+
+void apply_inv_sbox(uint64_t *const state);
+
+void exp_acc(const uint64_t m, const uint64_t *base,
+             const uint64_t *tail, uint64_t *const res);
```

Comment:
I think if we change the order of functions below so that a function is not called before its implementation is available (e.g., move `apply_round` after `apply_mds` etc.), we can get rid of these definitions.